src/Mavi.jl

```diff
@@ -52,9 +52,31 @@ struct System{T, C1<:SpaceCfg, C2<:DynamicCfg}
     num_p::Int
 end
 
+"""
+Checks if all particles are inside the given configuration space. If not, throws an error.
+"""
+function check_inside(state::State, space_cfg::RectangleCfg)
+    if all(x -> 0.0 <= x <= space_cfg.length, state.x) && all(y -> 0.0 <= y <= space_cfg.height, state.y)
+        return true
+    else
+        return false
+    end
+end
+function check_inside(state::State{T}, space_cfg::CircleCfg) where {T}
+    r2 = state.x.^2 + state.y.^2
+    if all(r -> r <= space_cfg.radius^2, r2)
+        return true
+    else
+        return false
+    end
+end
+
+
 function System(;state::State{T}, space_cfg, dynamic_cfg, int_cfg) where {T}
+    if check_inside(state,space_cfg) == false
+        throw("particles must be inside space")
```

Comment:
Acho que uma melhor mensagem de erro é "Particle with id=$(outside_id) is out of bounds."